components/Buttons/MarkPlayedButton.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <v-btn v-if="canMarkWatched" icon dark @click.stop.prevent="togglePlayed">
+    <v-icon :color="isPlayed ? 'primary' : 'white'"> mdi-check </v-icon>
+  </v-btn>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapActions } from 'vuex';
+import { BaseItemDto } from '@jellyfin/client-axios';
+
+export default Vue.extend({
+  props: {
+    item: {
+      type: Object as () => BaseItemDto,
+      required: true
+    }
+  },
+  data() {
+    return {
+      isPlayed: false
+    };
+  },
+  computed: {
+    canMarkWatched(): boolean {
+      if (
+        ['Series', 'Season', 'BoxSet', 'AudioPodcast', 'AudioBook'].includes(
+          this.item.Type || ''
+        )
+      ) {
+        return true;
+      }
+
+      if (this.item.MediaType === 'Video' && this.item.Type !== 'TvChannel') {
+        return true;
+      }
+
+      return false;
+    }
+  },
+  created() {
+    this.isPlayed = this.item.UserData?.Played || false;
+  },
+  methods: {
+    ...mapActions('snackbar', ['pushSnackbarMessage']),
+    async togglePlayed(): Promise<void> {
+      try {
+        if (this.isPlayed) {
+          await this.$api.playState.markUnplayedItem({
+            userId: this.$auth.user.Id,
+            itemId: this.item.Id || ''
+          });
+          this.isPlayed = false;
+        } else {
+          await this.$api.playState.markPlayedItem({
+            userId: this.$auth.user.Id,
+            itemId: this.item.Id || ''
+          });
+          this.isPlayed = true;
+        }
+      } catch (error) {
+        this.pushSnackbarMessage({
+          message: this.$t('unableToTogglePlayed'),
```

Comment:
Can we put all the snackbar messages under a ``snackbar`` subkey? Just do it for this one but let's make another PR to cleanup the rest of them.

Reply:
Sounds good, I'll do it in a seperate PR though